crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2235,24 +2235,18 @@ impl Room {
             return Ok(None);
         };
 
-        let room_power_levels = if let Some(event) = self
+        let power_levels = self
             .get_state_event_static::<RoomPowerLevelsEventContent>()
             .await?
             .and_then(|e| e.deserialize().ok())
-        {
-            event.power_levels()
-        } else {
-            return Ok(None);
-        };
+            .map(|e| e.power_levels().into());
```

Comment:
I think this change is wrong: we should only return the push context when it's complete (that's what the function's signature says), otherwise we can't properly compute notifications later on.

Reply:
What if the user wants to use this method with an invited room? The power levels won't be available.

Reply:
Ah cool. I previously read Ruma's code and it did require the power levels, but now I see you [made](https://github.com/ruma/ruma/commit/90d3605b87e761e9c89125bc3ef4d7fff7e29b68) it optional there. Good stuff :+1: